aws/rust-runtime/aws-credential-types/src/lib.rs

```diff
@@ -67,13 +81,22 @@ construct credentials from hardcoded values.
 //! }
 //! ```
 
+mod cache;
+mod credential_fn;
 mod credentials_impl;
+pub mod lazy_caching;
 mod provider;
+mod time_source;
 
+pub use cache::ExpiringCache;
+pub use credential_fn::provide_credentials_fn;
 pub use credentials_impl::Credentials;
+pub use lazy_caching::LazyCachingCredentialsProvider;
 pub use provider::error;
 pub use provider::error::CredentialsError;
 pub use provider::future;
 pub use provider::ProvideCredentials;
 pub use provider::Result;
 pub use provider::SharedCredentialsProvider;
+pub use time_source::ManualTimeSource;
+pub use time_source::TimeSource;
```

Comment:
I think we should consider re-exporting less at the top-level.

Reply:
Absolutely. Reduced the number of re-exports in a1f1f78.